book/doctrine.rst

```diff
@@ -916,8 +924,81 @@ a ``name`` field and the associated getter and setter functions.
 Relationship Mapping Metadata
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To relate the ``Category`` and ``Product`` entities, start by creating a
-``products`` property on the ``Category`` class:
+In our example, each category can be associated with *many* products, while
+each product can be associated with only *one* category. This relationship
+can be summarized as: *many* products to *one* category (or equivalently,
+*one* category to *many* products).
+
+From the perspective of the ``Product`` entity, this is a many-to-one relationship.
+From the perspective of the ``Category`` entity, this is a one-to-many relationship.
+This is important, because the relative nature of the relationship determines
+which mapping metadata to use. It also determines which class *must* hold
+a reference to the other class.
+
+To relate the ``Product`` and ``Category`` entities, simply create a ``category``
+property on the ``Product`` class, annotated as follows:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/AppBundle/Entity/Product.php
+
+        // ...
+        class Product
+        {
+            // ...
+
+            /**
+             * @ORM\ManyToOne(targetEntity="Category", inversedBy="products")
+             * @ORM\JoinColumn(name="category_id", referencedColumnName="id")
```

Comment:
While the annotation is admittedly optional in this situation, I think it should be maintained because it makes the code more expressive.  To a beginner who might have trouble imagining which of the two entities actually holds the foreign key, the `@ORM\JoinColumn` quickly answers that question (since it exists only in ``Product``, but not ``Category``).

The presence of this annotation also facilitates an easy discussion of the `category_id` column further down in the tutorial.  In the absence of `@ORM\JoinColumn`, we'd have to then veer into an explanation of how Doctrine automatically determines the name of the foreign key field, and how Doctrine knows in which of the two tables the foreign key should be stored.